routing/index_graph.cpp

```diff
@@ -1,9 +1,48 @@
-#include "index_graph.hpp"
+#include "routing/index_graph.hpp"
+
+#include "routing/restrictions_serialization.hpp"
 
 #include "base/assert.hpp"
 #include "base/exception.hpp"
 
-#include "std/limits.hpp"
+#include <algorithm>
+#include <limits>
+
+namespace
+{
+using namespace routing;
+using namespace std;
+
+bool IsRestricted(RestrictionVec const & restrictions, Segment const & u, Segment const & v, bool isOutgoing)
+{
+  uint32_t const featureIdFrom = isOutgoing ? u.GetFeatureId() : v.GetFeatureId();
+  uint32_t const featureIdTo = isOutgoing ? v.GetFeatureId() : u.GetFeatureId();
```

Comment:
If my conjecture is true, we still need a bit indicating what parts of features (starts or ends) participate in a restriction, regardless of the type of the restriction(No or Only). But the code is way much simper that the similar one on the master. Whatever, from my side, this comment and a comment about complexity of LastJointId are the most important, for other comments it would be nice if you'll resolve them, but not critical.